pkg/auth/auth.go

```diff
@@ -0,0 +1,162 @@
+package auth
+
+import (
+	sha "crypto/sha512"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/casbin/casbin/v2"
+	"github.com/casbin/casbin/v2/model"
+)
+
+type authRecord struct {
+	role   string
+	expiry time.Time
+}
+
+type apiKeys map[string]authRecord
+
+type Authenticator struct {
+	user, pass string
+	apiKeys    apiKeys
+	enforcer   *casbin.Enforcer
+}
+
+func New() (*Authenticator, error) {
+	m, err := model.NewModelFromString(`
+	[request_definition]
+	r = sub, obj, act
+	
+	[policy_definition]
+	p = sub, obj, act
+	
+	[policy_effect]
+	e = some(where (p.eft == allow))
+	
+	[matchers]
+	m = r.sub == p.sub && keyMatch(r.obj, p.obj) && regexMatch(r.act, p.act)`)
+
+	if err != nil {
+		return nil, err
+	}
+
+	e, err := casbin.NewEnforcer(m)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := applyPolicies(e); err != nil {
+		return nil, err
+	}
+
+	auth := Authenticator{
+		user:     os.Getenv("AUTH_USERNAME"),
+		pass:     os.Getenv("AUTH_PASSWORD"),
+		apiKeys:  make(apiKeys),
+		enforcer: e,
+	}
+
+	return &auth, nil
+}
+
+func (a *Authenticator) Authorize(u, p string) bool {
+	return a.user == u && a.pass == p
```

Comment:
I am suggesting to have the password hashed and checked with, for example, bcrypt.GenerateFromPassword.